0.0.78.bump-grafana/pkg/grafana/api.go

```diff
@@ -0,0 +1,104 @@
+package grafana
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/oslokommune/okctl-upgrade/0.0.78.bump-grafana/pkg/commonerrors"
+	"github.com/oslokommune/okctl-upgrade/0.0.78.bump-grafana/pkg/logger"
+)
+
+// Upgrader is a sample okctl component
+type Upgrader struct {
+	logger  logger.Logger
+	dryRun  bool
+	confirm bool
+}
+
+// Upgrade upgrades the component
+func (c Upgrader) Upgrade() error {
+	c.logger.Info("Upgrading Grafana")
+
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+	if kubeconfigPath == "" {
+		return errors.New("missing required KUBECONFIG environment variable")
+	}
+
+	kubectlClient, err := acquireKubectlClient(kubeconfigPath)
+	if err != nil {
+		return fmt.Errorf("acquiring kubectl client: %w", err)
+	}
+
+	err = preflight(c.logger, kubectlClient)
+	if err != nil {
+		if errors.Is(err, ErrNothingToDo) {
+			return nil
+		}
+
+		return fmt.Errorf("running preflight checks: %w", err)
+	}
+
+	c.logger.Debug(fmt.Sprintf("Passed preflight test. Upgrading Grafana to %s", targetGrafanaVersion.String()))
+
+	if !c.dryRun && !c.confirm {
+		c.logger.Info(fmt.Sprintf("This will bump Grafana to %s", targetGrafanaVersion.String()))
+
+		answer, err := c.askUser("Do you want to proceed?")
+		if err != nil {
+			return fmt.Errorf("prompting user: %w", err)
+		}
+
+		if !answer {
+			return commonerrors.ErrUserAborted
+		}
+	}
```

Comment:
I suggest we delete this. Prompting the user is something we only want to do for potentional destructive stuff like deleting logs. Also, since upgrades are simulated fist, we're already prompting the user if they're sure they want to upgrade.